interfaces.md

```diff
@@ -0,0 +1,73 @@
+# Interfaces
+
+Document that outlines what each layer requires and provides in more detail.
+
+Goal of this document: provide a clear interface for other teams working on other parts of the stack. Specifically: Validity Labs and Nym, working on mixnet designs at transport privacy layer.
+
+## Data Sync clients
+
+TODO: Later
+
+Data sync clients are...
+
+They require... (reliable transmission of data between pairs of peers or groups of peers)
+
+They provide ... (end user functionality like group chat, 1:1 chat, other forms of socio-economic coordination mechanisms)
+
+## Data sync
+
+The data sync layer is responsible for synchronizing data across multiple devices.
+
+It requires a transport layer security protocol that is able to deliver data securely from one device to another. It may also deliver from one device to many other devices. Data can be delayed, lost, reordered or duplicated. The transport layer security protocol is responsible for properties such as confidentiality, integrity, authenticity and forward secrecy. This layer does not care about what underlying transport is used.
+
+It also requires that messages are immutable/idempotent, i.e. they can be referred to by some cryptographic hash.
+
+It provides reliable data synchronization with casual consistency guarantees in a specific group text without relying on centralized services.
+
+(Inspired by https://code.briarproject.org/briar/briar-spec/blob/master/protocols/BSP.md)
+
+## Secure transport
+
+TODO: Connection-oriented vs connectionless, make ths imore explicit
```

Comment:
`make ths imore explicit` -> `make this more explicit`

Reply:
fixed